keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1475,225 +1382,382 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
+// the result of that was.
+type ResourceClaimStatus struct {
+    // Allocation is set once the claim has been allocated successfully.
+    // +optional
+    Allocation *AllocationResult
+
+    // ReservedFor indicates which entities are currently allowed to use
+    // the claim. A Pod which references a ResourceClaim which is not
+    // reserved for that Pod will not be started. A claim that is in
+    // use or might be in use because it has been reserved must not get
+    // deallocated.
+    //
+    // In a cluster with multiple scheduler instances, two pods might get
+    // scheduled concurrently by different schedulers. When they reference
+    // the same ResourceClaim which already has reached its maximum number
+    // of consumers, only one pod can be scheduled.
     //
-    // In addition, for each type in NamedResourcesAttributeValue there is a map that
-    // resolves to the corresponding value of the instance under evaluation.
-    // For example:
+    // Both schedulers try to add their pod to the claim.status.reservedFor
+    // field, but only the update that reaches the API server first gets
+    // stored. The other one fails with an error and the scheduler
+    // which issued it knows that it must put the pod back into the queue,
+    // waiting for the ResourceClaim to become usable again.
+    //
+    // There can be at most 32 such reservations. This may get increased in
+    // the future, but not reduced.
     //
-    //    attributes.quantity["a"].isGreaterThan(quantity("0")) &&
-    //    attributes.stringslice["b"].isSorted()
-    Selector string
+    // +listType=map
+    // +listMapKey=uid
+    // +patchStrategy=merge
+    // +patchMergeKey=uid
+    // +optional
+    ReservedFor []ResourceClaimConsumerReference
 }
 ```
 
-NamedResourcesFilter and NamedResourcesRequest currently have the same
-content. Despite that, they are defined as separate structs because that might
-change in the future.
+##### DeviceClass
+
+```go
+// DeviceClass is a vendor or admin-provided resource that contains
+// device configuration and selectors. It can be referenced in
+// the device requests of a claim to apply these presets.
+// Cluster scoped.
+type DeviceClass struct {
+    metav1.TypeMeta
+    // Standard object metadata
+    // +optional
+    metav1.ObjectMeta
+
+    // Each selector must be satisfied by a device which is claimed via this class.
+    //
+    // +optional
+    // +listType=atomic
+    Selectors []Selector
 
+    // Config defines configuration parameters that apply to each device that is claimed via this class.
+    // Some classses may potentially be satisfied by multiple drivers, so each instance of a vendor
+    // configuration applies to exactly one driver.
+    //
+    // They are passed to the driver, but are not consider while allocating the claim.
+    //
+    // +optional
+    // +listType=atomic
+    Config []ConfigurationParameters
```

Comment:
This is the same struct used in the claim, and now includes `RequestNames`, which has no meaning in the Class. Should we have a different type for the class config parameters?

Reply:
Maybe? It's not a problem... until it becomes a problem :grin:

As it's all in ResourceSlice and very localized, I think a single struct is okay.